### ISSUE TITLE
Forgot password requirement 2.5.6 is confusing

### ISSUE THREAD

**Issue by tghosth** [OWASP/ASVS#2457]
We currently have the following requirement for 2.5.6.

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **2.5.6** | [GRAMMAR] Verify that forgotten password, and other recovery paths use a secure recovery mechanism, such as time-based OTP (TOTP) or other soft token, mobile push, or another offline recovery mechanism. | ✓ | ✓ | ✓ | 640 |
| **2.5.7** | [GRAMMAR, LEVEL L2 > L1] Verify that if OTP or other multi-factor authentication factors are lost, that evidence of identity proofing is performed at the same level as during enrollment. | ✓ | ✓ | ✓ | 308 |

To be honest, I find it confusing. We want a secure password recovery mechanism but you don't automatically use a TOTP or something for a regular password reset. 

I think I would reword and simplify to:

"_[GRAMMAR] Verify that a secure forgotten password reset process is implemented that does not bypass any enabled multi-factor authentication mechanisms._"

The references contain a link to the forgot password cheatsheet.

Thoughts @jmanico @elarlang @ryarmst @randomstuff @TobiasAhnoff ?

**Comment by tghosth** [comment]
> "[GRAMMAR] Ensure that the secure password reset process enforces the use of multi-factor authentication if it is enabled."

> Verify that a secure forgotten password reset process is implemented that does not bypass any enabled multi-factor authentication mechanisms.

I'm a bit concerned that both these proposals bury the point that it is important to have a secure forgot password process and that the MFA point is an additional to this.

**Comment by ryarmst** [comment]
@tghosth Was the primary intent of 2.5.6 to mitigate account takeover through email hijacking (via subsequent password reset)? I think mandating MFA is probably a necessary component of this. It could be abstracted more, but I think it might then be unclear with respect to what is required to be implemented. 

Are we re-examining 2.5.7?

**Comment by randomstuff** [comment]
(Adding MFA is good and all. On the other side, enforcing more MFA everywhere (combined with automatic account lockdown after suspicious activity detection based on some machine learning model) is a very good way to make the probability that the normal people looses access to all his accounts tend to 1 as time goes by. :smile:)

**Comment by tghosth** [comment]
> @tghosth Was the primary intent of 2.5.6 to mitigate account takeover through email hijacking (via subsequent password reset)? I think mandating MFA is probably a necessary component of this. It could be abstracted more, but I think it might then be unclear with respect to what is required to be implemented.

I think the primary intent is to have a secure forgot password process. Email hijacking will certainly be hard to mitigate in a single-factor auth situation. This is why the in addition to the primary intent, my suggestion clarifies that this should not bypass MFA.

> Are we re-examining 2.5.7?

I think 2.5.7 is ok and I brought it in to explain why I think we are already covered on MFA and that we don't want use of MFA to confuse 2.5.6.

> (Adding MFA is good and all. On the other side, enforcing more MFA everywhere (combined with automatic account lockdown after suspicious activity detection based on some machine learning model) is a very good way to make the probability that the normal people looses access to all his accounts tend to 1 as time goes by. 😄)

This is part of the reason that I want 2.5.6 to be simple and clear.

**Comment by tghosth** [comment]
@elarlang @jmanico @ryarmst @randomstuff any further comments on what I wrote above, as a reminder, I still propose:

"_[GRAMMAR] Verify that a secure forgotten password reset process is implemented that does not bypass any enabled multi-factor authentication mechanisms._"

**Comment by jmanico** [comment]
How about a positive take on this:

_[MODIFIED] Verify that a secure forgotten password reset process is implemented so that it includes a multi-factor authentication mechanism._

**Comment by tghosth** [comment]
> [MODIFIED] Verify that a secure forgotten password reset process is implemented so that it includes a multi-factor authentication mechanism.

I don't think that we can mandate that a forgot password actively includes MFA, I would rather keep the topics separate. This is intended to be a simple requirement which basically says follow the cheatsheet and don't bypass MFA if it is enabled. Elsewhere we will talk about the importance of MFA. I am going to open a PR.

**Comment by tghosth** [comment]
Opened #2480

**Comment by elarlang** [comment]
> forgotten password reset process

It can be read as "a process about password reset that is forgotten".

**Comment by tghosth** [comment]
"_[MODIFIED] Verify that a secure process for resetting a forgotten password is implemented, that does not bypass any enabled multi-factor authentication mechanisms._"

**Comment by elarlang** [comment]
We can add brackets a well.
> Verify that a secure process for resetting a (forgotten) password is implemented, that does not bypass any enabled multi-factor authentication mechanisms.

With or without brackets, please update the PR.
